tests/postgres_tests/test_array.py

```diff
@@ -466,8 +466,8 @@ def test_group_by_order_by_select_index(self):
                 ],
             )
         sql = ctx[0]["sql"]
-        self.assertIn("GROUP BY 1", sql)
-        self.assertIn("ORDER BY 1", sql)
+        self.assertIn("GROUP BY 2", sql)
+        self.assertIn("ORDER BY 2", sql)
```

Comment:
The order of the columns in the select was non-deterministic, since it was linked to the resulting order of the set: `self.annotation_select_mask.union(names)`.